### PR TITLE
LPD-15081 Handle the iframe nodes in the content

### PR DESCRIPTION
**Pull request by robertoDiaz**
Please check the LPP or contact with me if you need context about this.

### CODE REVIEW

File: modules/apps/portal-security/portal-security-iframe-sanitizer-test/src/testIntegration/java/com/liferay/portal/security/iframe/sanitizer/internal/test/IframeSanitizerImplTest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testSanitizeHTMLWithIFrame() throws Exception {
+		Assert.assertEquals(
+			_BASIC_HTML_CONTENT + _EXPECTED_IFRAME_TAG,
+			_iFrameSanitizer.sanitize(
+				0, 0, 0, "", 0, ContentTypes.TEXT_HTML, new String[0],
+				_BASIC_HTML_CONTENT + _INITIAL_IFRAME_TAG, new HashMap<>()));
```
Comment (marcogalluzzi) [inline modules/apps/portal-security/portal-security-iframe-sanitizer-test/src/testIntegration/java/com/liferay/portal/security/iframe/sanitizer/internal/test/IframeSanitizerImplTest.java:43]: I would create and reuse a method like `private String _sanitize(String content, String contentType)` calling `_iFrameSanitizer.sanitize()` with all other parameters with fixed values.

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]
It looks very good to me. Just some minor changes (naming and private method creation).

Should this work also be reviewed by the team owner of `portal-security` module?

File: modules/apps/portal-security/portal-security-iframe-sanitizer-test/src/testIntegration/java/com/liferay/portal/security/iframe/sanitizer/internal/test/IframeSanitizerImplTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+				_BASIC_CONTENT, new HashMap<>()));
+
+		Assert.assertEquals(
+			_BASIC_CONTENT,
+			_iFrameSanitizer.sanitize(
+				0, 0, 0, "", 0, ContentTypes.TEXT_PLAIN, new String[0],
+				_BASIC_HTML_CONTENT, new HashMap<>()));
+	}
+
+	private void _testWithModifiedConfiguration(
```
[the remark this answers is not in the document]
Comment (robertoDiaz) [reply modules/apps/portal-security/portal-security-iframe-sanitizer-test/src/testIntegration/java/com/liferay/portal/security/iframe/sanitizer/internal/test/IframeSanitizerImplTest.java:112]: I'll change it.

**Comment by robertoDiaz** [comment]
Updated

File: modules/apps/portal-security/portal-security-iframe-sanitizer/build.gradle [hunk 3]
```diff
@@ -1,4 +1,5 @@
 dependencies {
 	compileOnly group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "default"
+	compileOnly group: "org.jsoup", name: "jsoup", version: "1.15.3"
```
Comment (AliciaGarciaGarcia) [inline modules/apps/portal-security/portal-security-iframe-sanitizer/build.gradle:4]: sorry for the noise, but a question:

Didn't we want to get rid of this lib?

Comment (robertoDiaz) [reply modules/apps/portal-security/portal-security-iframe-sanitizer/build.gradle:4]: no.

In fact @adolfopa suggested me to use it

**Comment by robertoDiaz** [comment]
I'll move this forward, if you don't mind

**Comment by robertoDiaz** [comment]
ci:forward

**Comment by marcogalluzzi** [comment]
> Updated

Did you push a new commit?

**Comment by robertoDiaz** [comment]
No, same commit to avoid noise with the requested changes.

I've also use the "withconfiguration"method in all cases (to avoid errors due configuration changes)

**Comment by robertoDiaz** [comment]
Found a small problem with current implementation. Fixing it.

**Comment by robertoDiaz** [comment]
ci:forward

File: modules/apps/portal-security/portal-security-iframe-sanitizer/build.gradle (hunk 3 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/portal-security/portal-security-iframe-sanitizer/build.gradle:4]: :D 

perfect then!

**Comment by robertoDiaz** [comment]
ci:test:relevant

**Comment by robertoDiaz** [comment]
ci:report:2532636423

**Comment by robertoDiaz** [comment]
ci:report:2532805647

**Comment by robertoDiaz** [comment]
Broken tests seem unrelated

**Comment by robertoDiaz** [comment]
ci:forward:force

**Comment by marcogalluzzi** [comment]
ci:forward:force

**Pull request closed without merging**